### ISSUE TITLE
How to specify which dimension to reduce along when using `sum` ? 

### ISSUE THREAD

**Issue by mspronesti** [LaurentMazare/tch-rs#613]
Basically, I want to convert `torch.sum(tensor, 1)`. `tch::Tensor` has a `sum(dtype)` method which doesn't unfortunately allow to specify which dimension to reduce: it reduces along all of them, like `torch.sum` does when no `dim` is passed.

I guess the correct method to call is `sum_dim_intlist`, however `dim` must implement `Into<Option<&'a [i64]>>`, which forces one to use a slice. Isn't there anything simpler than this ? 

```rs
some_tensor.sum_dim_intlist([1].as_slice(), false, Kind::Int64)
```

**Comment by LaurentMazare** [comment]
I think you're right and that's the appropriate way to do it, I wouldn't know about a simpler way now but I've just prototype in #621 a way to simplify this and allow one to write the following:
```rust
some_tensor.sum_dim_intlist(1, false, Kind::Int64)
```
However this is a bit of a breaking change so we probably want to think a bit more about it (besides only making it a major revision change which is already the case).

**Comment by LaurentMazare** [comment]
I've merged these changes via #682 so the following should work now.
```rust
some_tensor.sum_dim_intlist(1, false, Kind::Int64)
```
Hopefully this makes it simpler to write code that involve reduction across a dimension. Feel free to re-open if you run into further issues.
